### PR TITLE
feat: product variants

### PR DESCRIPTION
**Pull request by Ianmohc**

### CODE REVIEW

File: shop.go [hunk 1]
```diff
@@ -12,4 +12,6 @@ type Shop interface {
 	Fulfillments() FulfillmentRepository
 	// FulfillmentEvents represent tracking events that belong to a fulfillment of one or more items in an order.
 	FulfillmentEvents() FulfillmentEventRepository
+	// Variants represents options or versions of that make up an individual product.
```
Comment (harry-mohc) [inline shop.go:15]: comment isn't grammatically correct

File: variants.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package shopify
+
+import "time"
+
+// Variants is a collection of variants
+type Variants []Variant
+
+// Variant is a version of a `Product` for a shop.
```
Comment (harry-mohc) [inline variants.go:8]:
```suggestion
// Variant is a specific version of a product
```

File: variants.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+	Barcode string
+	// CreatedAt is the date and time when the product variant was created.
+	CreatedAt time.Time
+	// CreatedAt is the date and time when the product variant was last modified.
+	UpdatedAt time.Time
+}
+
+// VariantRepository maintains the product variants of a shop.
+type VariantRepository interface {
+	// Get gets an order
```
Comment (harry-mohc) [inline variants.go:30]: it does not

File: variants.go [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+	Get(SKU string) (Variant, error)
```
Comment (harry-mohc) [inline variants.go:31]: no for shopify you get by its ID, getting a product by its SKU is ecommerce not shopify, if you could do this with shopify we wouldn't need ecommerce

File: variants.go [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+	// Barcode is the barcode, UPC, or ISBN number for the product.
+	Barcode string
+	// CreatedAt is the date and time when the product variant was created.
+	CreatedAt time.Time
+	// CreatedAt is the date and time when the product variant was last modified.
+	UpdatedAt time.Time
+}
+
+// VariantRepository maintains the product variants of a shop.
+type VariantRepository interface {
```
Comment (harry-mohc) [inline variants.go:29]: create a variant-errors.go file and create an ErrVariantNotFound error

**Review by harry-mohc — CHANGES_REQUESTED** [review]

File: variants.go (hunk 3 above)
Comment (Ianmohc) [reply variants.go:30]: You can see I copied that 🤕

**Review by harry-mohc — APPROVED** [review]

**Pull request merged**